test/Grpc.NetCore.HttpClient.Tests/AsyncServerStreamingCallTests.cs

```diff
@@ -176,5 +169,34 @@ public void ClientStreamReader_WriteWithInvalidHttpStatus_ErrorThrown()
 
             Assert.AreEqual("Bad gRPC response. Expected HTTP status code 200. Got status code: 404", ex.Message);
         }
+
+        [Test]
+        public async Task AsyncServerStreamingCall_NoContent_TrailersReturnedWithHeaders()
```

Comment:
let's add a negative test case where grpc-status is added to the headers, but there's also a payload and actual trailers (where grpc-status is missing) and verify that scenario fails.